language/move-vm/runtime/src/session.rs

```diff
@@ -55,18 +55,61 @@ impl<'r, 'l, R: RemoteCache> Session<'r, 'l, R> {
         )
     }
 
+    /// Execute a Move script function with the given arguments.
+    ///
+    /// Unlike `execute_function` which is designed for system logic, `execute_script_function` is
+    /// mainly designed to call a script function in an existing module. It similar to
+    /// `execute_script` except that execution of the "script" begins with the specified function
+    ///
+    /// The Move VM MUST return a user error (in other words, an error that's not an invariant
+    /// violation) if
+    ///   - The function does not exist
+    ///   - The function does not have a script-visible
+    ///   - Type arguments refer to a non-existent type.
+    ///   - Arguments (senders included) are invalid or fail to match the signature of the script.
```

Comment:
Might make sense to mention "invalid" or "fail to match" on separate lines + expand on what `invalid` means a bit (possibly including pointer to relevant bytecode verifier pass), since the rules are not necessarily obvious

Reply:
I updated the comments for this and script. Let me know what you think